src/Map.js

```diff
@@ -4,13 +4,81 @@ import mapboxgl from 'mapbox-gl';
 
 import Popup from "Popup";
 import "Map.scss"
-import points from "points.json";
+
 import token from "token.json";
+import points from "points.json";
+import markers from "markers.json";
 
 const PORTO_LNG = -8.6291;
 const PORTO_LAT = 41.1579;
 const ZOOM = 13;
 
+let loadMarkers = async (map, markers) => {
+  return Promise.all(
+    Object.entries(markers).map(([marker, filename]) =>
+      new Promise((resolve, reject) => {
+        map.loadImage(filename, (error, image) => {
+          if (error) {
+            reject(error);
+            return;
+          }
+
+          map.addImage(marker, image);
+          resolve();
+        })
+      })
+    )
+  )
+};
+
+const loadPoints = (map, points) => {
+  map.addSource("points", points)
+
+  map.addLayer({
+    id: "points",
+    type: "symbol",
+    source: "points",
+    layout: {
+      "icon-image": "{icon}",
```

Comment:
This `"{icon}"` thing was all the magic that was needed to render different images per icon the same layer.